test/e2e/kubelet_deletes.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// Interval to poll /runningpods on a node
+	pollInterval = 1 * time.Second
+)
+
+// waitTillKubeletDeletesPods waits till pods with UID in targetPodUIDs set
+// are visible on the kubelet's /runningpods endpoint.
+func waitTillKubeletDeletesPods(restClient *client.Client, targetPodUIDs util.StringSet, nodeName string, timeout time.Duration) error {
+	return wait.Poll(pollInterval, timeout, func() (bool, error) {
+		pods, err := GetKubeletPods(restClient, nodeName)
+		if err != nil {
+			return false, err
+		}
+
+		alive := 0
+		for _, pod := range pods.Items {
+			if targetPodUIDs.Has(string(pod.UID)) {
+				alive += 1
+			}
+		}
+		if alive > 0 {
+			Logf("Waiting on kubelet to delete %d pods on node %q", alive, nodeName)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+var _ = Describe("Clean up pods on node", func() {
+	var numNodes int
+	framework := NewFramework("kubelet-delete")
+
+	BeforeEach(func() {
+		nodes, err := framework.Client.Nodes().List(labels.Everything(), fields.Everything())
+		expectNoError(err)
+		numNodes = len(nodes.Items)
+	})
+
+	type DeleteTest struct {
+		podsPerNode int
+		timeout     time.Duration
+	}
+
+	deleteTests := []DeleteTest{
+		{podsPerNode: 10, timeout: 3 * time.Minute},
```

Comment:
No, at least not on my GCE cluster. However, this is more of a correctness than a performance test. The timeout is long enough that i'd not fail randomly due to other issues, but it will still surface any kind of hang in kubelet. Plus, with a slightly longer timeout, we can actually print out the latency in the end (without failing prematurely).